### DISCUSSION TITLE
Route match by header regex

### DISCUSSION THREAD

**Discussion by vanhtuan0409** [Kong/kong#7300] [Ideas and feature requests]
Since 1.3, It is possible to match a route by request header but it requires an exact match.
Is it possible to add regex support for header routing :D 
https://docs.konghq.com/2.0.x/proxy/#request-headers-except-host

**Comment by vanhtuan0409** [comment]
added a PR for this issue. Can you take time to have a look at this @bungle :D ?

**Comment by carlosrmendes** [comment]
any updates for this feature?

**Comment by kikito** [comment]
This feature was included in kong 2.8

**Reply by vagharsh** [reply]
can you please provide a documentation link on how to use the regex ?

**Reply by chronolaw** [reply]
This is it: https://docs.konghq.com/gateway/2.8.x/admin-api/#route-object, but I did not find the description about regex header too.

**Reply by chronolaw** [reply]
For regex header it should have a prefix `~*`, like `"~*xxx"`.

**Comment by vagharsh** [comment]
now i have an issue with routing the traffic based on the User-Agent header
in this annotation i am using and it's not working.
```
      annotations:
        konghq.com/headers.user-agent: "~*.*Example\-Foo\-Bar\:.*"
```

**Reply by vagharsh** [reply]
apparently it doesn't recognize capital letters .. since when i changes the rule to the following it worked : 
```  annotations:
        konghq.com/headers.user-agent: "~*.*example\-foo\-bar\:.*"
```
and also 
```  annotations:
        konghq.com/headers.user-agent: "~*.*example\-foo\-bar\:"
```

**Reply by chronolaw** [reply]
The regex matching is case sensitive.

**Reply by vagharsh** [reply]
then why it was not matching the user-agent header value "Example-Foo-Bar" with this rule
```
      annotations:
        konghq.com/headers.user-agent: "~*.*Example\-Foo\-Bar\:.*"
```
that is weird

**Reply by chronolaw** [reply]
Perhaps we should use `"~*.*Example\-Foo\-Bar\:?.*"` (add a `?` after `:`).
